alert-common/src/main/java/com/synopsys/integration/alert/common/action/api/AbstractJobResourceActions.java

```diff
@@ -0,0 +1,216 @@
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import org.springframework.http.HttpStatus;
+
+import com.synopsys.integration.alert.common.action.ActionResponse;
+import com.synopsys.integration.alert.common.action.ValidationActionResponse;
+import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
+import com.synopsys.integration.alert.common.enumeration.DescriptorType;
+import com.synopsys.integration.alert.common.exception.AlertDatabaseConstraintException;
+import com.synopsys.integration.alert.common.persistence.accessor.DescriptorAccessor;
+import com.synopsys.integration.alert.common.persistence.model.RegisteredDescriptorModel;
+import com.synopsys.integration.alert.common.rest.model.FieldModel;
+import com.synopsys.integration.alert.common.rest.model.JobFieldModel;
+import com.synopsys.integration.alert.common.rest.model.ValidationResponseModel;
+import com.synopsys.integration.alert.common.security.authorization.AuthorizationManager;
+
+public abstract class AbstractJobResourceActions implements JobResourceActions, ValidateAction<JobFieldModel>, TestAction<JobFieldModel> {
+    private static final EnumSet<DescriptorType> ALLOWED_JOB_DESCRIPTOR_TYPES = EnumSet.of(DescriptorType.PROVIDER, DescriptorType.CHANNEL);
+    private AuthorizationManager authorizationManager;
+    private DescriptorAccessor descriptorAccessor;
+
+    public AbstractJobResourceActions(AuthorizationManager authorizationManager, DescriptorAccessor descriptorAccessor) {
+        this.authorizationManager = authorizationManager;
+        this.descriptorAccessor = descriptorAccessor;
+    }
+
+    protected abstract Optional<JobFieldModel> findJobFieldModel(UUID id);
+
+    protected abstract ActionResponse<JobFieldModel> createAfterChecks(JobFieldModel resource);
+
+    protected abstract ActionResponse<JobFieldModel> deleteAfterChecks(UUID id);
+
+    protected abstract ActionResponse<List<JobFieldModel>> readAllAfterChecks();
+
+    protected abstract ValidationActionResponse testAfterChecks(JobFieldModel resource);
+
+    protected abstract ActionResponse<JobFieldModel> updateAfterChecks(UUID id, JobFieldModel resource);
+
+    protected abstract ValidationActionResponse validateAfterChecks(JobFieldModel resource);
+
+    private Set<String> getDescriptorNames() {
+        Set<String> descriptorNames = Set.of();
+        try {
+            descriptorNames = descriptorAccessor.getRegisteredDescriptors()
+                                  .stream()
+                                  .filter(descriptor -> ALLOWED_JOB_DESCRIPTOR_TYPES.contains(descriptor.getType()))
+                                  .map(RegisteredDescriptorModel::getName)
+                                  .collect(Collectors.toSet());
+        } catch (AlertDatabaseConstraintException ex) {
+            // ignore or add a logger.
+        }
+
+        return descriptorNames;
```

Comment:
Instead of the Set object, could we return the statement in the try block otherwise return Set.of?